graph.cpp

```diff
@@ -26,9 +26,11 @@ Graph::Graph(uint64_t num_nodes): num_nodes(num_nodes) {
   bf = new BufferTree(buffer_loc_prefix, (1<<20), 8, num_nodes, GraphWorker::get_num_groups(), true);
   GraphWorker::start_workers(this, bf);
 #else
+  unsigned num_buckets = bucket_gen(num_nodes*num_nodes, 1); // TODO: don't fix num_bucket_factor
+  unsigned num_guesses = guess_gen(num_nodes*num_nodes);
   unsigned node_size = sizeof(Supernode) + double_to_ull(log2(num_nodes))*(
-        sizeof(Sketch) + bucket_gen(num_nodes*num_nodes, 1)*sizeof(vec_t) +
-        guess_gen(num_nodes)*sizeof(vec_hash_t));
+        sizeof(Sketch)
+        + num_buckets*num_guesses*(sizeof(vec_t) + sizeof(vec_hash_t)));
```

Comment:
Have you verified that this is reasonably close to the size we expect?

Reply:
Yes, testing with `n` up to 2^15, this size is between 3 times and 2 times as large as the naive `24*log_2^3(n)` estimate.